sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1325,6 +1325,53 @@ def test_gaussian_mixture_precisions_init_diag():
     )
 
 
+def test_gaussian_mixture_precisions_init():
+    """Non-regression test for #26415."""
+
+    def _generate_data(n_samples, n_features, n_components):
+        """Randomly generate samples and responsibilities"""
+        rs = np.random.RandomState(12345)
```

Comment:
The random state should be settable from the outside. Can you create a parameter to pass it?

Reply:
A `seed` parameter has been added to `_generate_data`.